cache/public/navigation_test.go

```diff
@@ -0,0 +1,91 @@
+package public
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ONSdigital/dp-frontend-search-controller/config"
+	"github.com/ONSdigital/dp-topic-api/models"
+	"github.com/ONSdigital/dp-topic-api/sdk"
+	apiError "github.com/ONSdigital/dp-topic-api/sdk/errors"
+	mockTopicCli "github.com/ONSdigital/dp-topic-api/sdk/mocks"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+var (
+	testNavData = &models.Navigation{
+		Description: "This is the top-level description.",
+		Links:       testTopicLinks,
+		Items:       testTopicNonReferential,
+	}
+
+	testNavDataCY = &models.Navigation{
+		Description: "This is the WELSH top-level description.",
+		Links:       testTopicLinks,
+		Items:       testTopicNonReferential,
+	}
+
+	testTopicLinks = &models.TopicLinks{
+		Self:      testLinkObject,
+		Subtopics: testLinkObject,
+		Content:   testLinkObject,
+	}
+
+	testLinkObject = &models.LinkObject{
+		HRef: "https://www.example.com",
+		ID:   "1234",
+	}
+
+	testTopicNonReferential = &[]models.TopicNonReferential{
+		{
+			Description: "This is an item description.",
+			Label:       "Item label",
+			Links:       testTopicLinks,
+			Name:        "Item name.",
+			Title:       "Item title.",
+			Uri:         "Item URI.",
+		},
+	}
+	cfg, _ = config.Get()
```

Comment:
You want to be calling this in each Test Function. And also, you don't want to omit the error even if it is used for testing purposes

Below `ctx := context.Background()` (line 56) in the func `TestUpdateNavigationData`, you should do

cfg, err := config.Get()
if err != nil {
     t.Errorf("failed to get config")
}

This will give the error and then fail the test

You should do this for each test function. If you are sharing configuration each test, it can led to a data race issue